File: src/main/java/nl/mvdr/adventofcode/adventofcode2024/day06/Part1.java
package nl.mvdr.adventofcode.adventofcode2024.day06;

import java.util.stream.Stream;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import nl.mvdr.adventofcode.solver.LongSolver;

public class Part1 implements LongSolver {

    private static final Logger LOGGER = LoggerFactory.getLogger(Part1.class);

    @Override
    public long solve(Stream<String> lines) {
        State initialState = State.parse(lines.toList());
        State endState = initialState.predictPath();
        return endState.visited()
                .stream()
                .map(GuardPosition::position)
                .distinct()
                .count();
    }

    public static void main(String[] args) {
        var instance = new Part1();

        var result = instance.solve("input-day06-2024.txt");

        LOGGER.info(result);
    }
}
 

File: adventofcode-2025/src/main/kotlin/nl/mvdr/adventofcode/adventofcode2025/day07/Part1.kt
package nl.mvdr.adventofcode.adventofcode2025.day07

import io.github.oshai.kotlinlogging.KotlinLogging
import nl.mvdr.adventofcode.solver.FunctionSolver

private val logger = KotlinLogging.logger{}

fun solvePart1(lines: Sequence<String>) = parseManifold(lines.toList()).countSplits()

fun main() {
    val result = FunctionSolver(::solvePart1).solve("input-day07.txt")
    logger.info { result }
}
